core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -696,6 +699,7 @@ private[spark] class ExecutorAllocationManager(
       allocationManager.synchronized {
         stageIdToNumTasks -= stageId
         stageIdToNumRunningTask -= stageId
+        activeStageIdToStageAttemptId -= stageId
```

Comment:
IIRC two stage attempts can actually be active at the same time, so this would not be correct.

Maybe @squito remembers this better than I do.